plugin/pkg/shoot/validator/admission_test.go

```diff
@@ -1562,6 +1562,303 @@ var _ = Describe("validator", func() {
+					Expect(coreInformerFactory.Core().InternalVersion().Seeds().Informer().GetStore().Add(&seed)).To(Succeed())
+				})
+
+				It("should allow creation of Shoot if reserved resources are less than allocatable resources", func() {
+					shoot.Spec.Provider.Workers = append(shoot.Spec.Provider.Workers, worker)
+
+					attrs := admission.NewAttributesRecord(&shoot, nil, core.Kind("Shoot").WithVersion("version"), shoot.Namespace, shoot.Name, core.Resource("shoots").WithVersion("version"), "", admission.Create, &metav1.CreateOptions{}, false, nil)
+
+					err := admissionHandler.Admit(context.TODO(), attrs, nil)
+
+					Expect(err).To(BeNil())
+				})
+
+				It("should not throw any errors if both global and worker kubeletConfigs are nil", func() {
+					worker.Kubernetes.Kubelet = nil
+					shoot.Spec.Provider.Workers = append(shoot.Spec.Provider.Workers, worker)
+
+					attrs := admission.NewAttributesRecord(&shoot, nil, core.Kind("Shoot").WithVersion("version"), shoot.Namespace, shoot.Name, core.Resource("shoots").WithVersion("version"), "", admission.Create, &metav1.CreateOptions{}, false, nil)
+
+					err := admissionHandler.Admit(context.TODO(), attrs, nil)
+
+					Expect(err).To(BeNil())
+				})
+
+				It("should not allow creation of Shoot if reserved CPU in the global kubeletConfig is more than allocatable CPU and worker kubeletConfig is nil", func() {
+					kubeletConfig.KubeReserved.CPU = &resourceCPU2
+					kubeletConfig.SystemReserved.CPU = &resourceCPU2
+					shoot.Spec.Kubernetes.Kubelet = kubeletConfig
+
+					worker.Kubernetes.Kubelet = nil
+					shoot.Spec.Provider.Workers = append(shoot.Spec.Provider.Workers, worker)
+
+					attrs := admission.NewAttributesRecord(&shoot, nil, core.Kind("Shoot").WithVersion("version"), shoot.Namespace, shoot.Name, core.Resource("shoots").WithVersion("version"), "", admission.Create, &metav1.CreateOptions{}, false, nil)
+
+					err := admissionHandler.Admit(context.TODO(), attrs, nil)
+
+					Expect(err).To(BeForbiddenError())
+					Expect(err.Error()).To(ContainSubstring("total reserved CPU (kubeReserved + systemReserved) cannot be more than the Node's allocatable CPU"))
+				})
+
+				It("should allow creation of Shoot if reserved CPU in the global kubeletConfig is more than allocatable CPU but the worker kubeletConfig have lesser reserved CPU", func() {
+					kubeletConfig.KubeReserved.CPU = &resourceCPU2
+					kubeletConfig.SystemReserved.CPU = &resourceCPU2
+					shoot.Spec.Kubernetes.Kubelet = kubeletConfig
+
+					shoot.Spec.Provider.Workers = append(shoot.Spec.Provider.Workers, worker)
+
+					attrs := admission.NewAttributesRecord(&shoot, nil, core.Kind("Shoot").WithVersion("version"), shoot.Namespace, shoot.Name, core.Resource("shoots").WithVersion("version"), "", admission.Create, &metav1.CreateOptions{}, false, nil)
+
+					err := admissionHandler.Admit(context.TODO(), attrs, nil)
+
+					Expect(err).To(BeNil())
```

Comment:
Any reason to not use

```suggestion
					Expect(err).ToNot(HaveOccurred())
```

?

Reply:
No reason, Just that the Admit function returns a nil if there isn't error. I can change it if you want.